test/parallel/test-stdin-require.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const execSync = require('child_process').execSync;
+const fs = require('fs');
+const path = require('path');
+
+common.refreshTmpDir();
+const filename = path.join(__dirname, '..', 'fixtures', 'baz.js');
+const out = path.join(common.tmpDir, 'js.out');
+const bin = process.execPath;
+const cmd = common.isWindows ?
+  `echo "require('${filename}')" | ${bin} > ${out} 2>&1 || more ${out}` :
+  `echo "require('${filename}')" | ${bin} &> ${out} || cat ${out}`;
+
+const result = execSync(cmd).toString();
+assert.strictEqual(result.includes('assertEncoding is not a function'),
```

Comment:
Is there anything we can explicitly detect? I'm just worried that if `assertEncoding` were to change names and a regression occurred, this wouldn't detect it.